### ISSUE TITLE
General Discussion

### ISSUE THREAD

**Issue by Rabenherz112** [awesome-selfhosted/awesome-selfhosted-data#33]
To avoid cluttering the Issues Tab with general discussions, questions, or other topics, I created a separate space to facilitate easier discussions on various subjects.

**Comment by Rabenherz112** [comment]
How exactly do you plan to handle the non-free page?
As far as I can see in the current preview (https://nodiscc.github.io/awesome-selfhosted-html-preview/#) there is no direct button. While these pages stay for now on GitHub as a Markdown file, or do you plan to also add another page for non-free entries?

What exactly is the Date meant to represent? I thought it was the initial addition time, but that didn't make any sense. Is this the latest update of the software? If yes, explaining that at the start page would probably make sense.

The Author Button (https://nodiscc.github.io/awesome-selfhosted-html-preview/#authors) downloads atm the Author File. O_O

Can't clone the repo normally, thanks to https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/software/diaspora*.yml
```
git clone https://github.com/awesome-selfhosted/awesome-selfhosted-data
Cloning into 'awesome-selfhosted-data'...
remote: Enumerating objects: 83823, done.
remote: Counting objects: 100% (83823/83823), done.
remote: Compressing objects: 100% (4599/4599), done.
remote: Total 83823 (delta 80702), reused 82271 (delta 79155), pack-reused 0
Receiving objects: 100% (83823/83823), 6.73 MiB | 21.33 MiB/s, done.
Resolving deltas: 100% (80702/80702), done.
error: invalid path 'software/diaspora*.yml'
fatal: unable to checkout working tree
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry with 'git restore --source=HEAD :/'
```

**Comment by nodiscc** [comment]
> How exactly do you plan to handle the non-free page? As far as I can see in the current preview (https://nodiscc.github.io/awesome-selfhosted-html-preview/#) there is no direct button. While these pages stay for now on GitHub as a Markdown file

The non-free page will only be rendered in the markdown-based list (there is a link to it in the header of https://nodiscc.github.io/awesome-selfhosted-html-preview). I don't want to mix Free/non-Free items, and the only way to keep them clearly separated with the HTML site would essentially be to duplicate the site, or duplicate pages. The markdown version will keep being updated, so this information stays available, but non-Free software has never been a major focus for this project (the project description itself states so). It is listed as a convenience, at the request of a few contributors/users.

> What exactly is the Date meant to represent? Is this the latest update of the software?

Yes. Looking at it now, I can see this is not very clear...

> If yes, explaining that at the start page would probably make sense.

What about using a tooltip when hovering the mouse over this `span` instead? You can see this mechanism is already used when hovering the `Anti-features` spans:

![image](https://github.com/awesome-selfhosted/awesome-selfhosted-data/assets/3950390/925a6f8d-0681-40ca-8677-ce0b83f9913f)

The tooltip would have the text `Date of last update` This would prevent adding too much information on-screen. Maybe dates rendered in range/red should have the tooltip `Date of last update - this project has not received any update since {6,12} months!` to clarify the purpose of the different background color.

> The Author Button (https://nodiscc.github.io/awesome-selfhosted-html-preview/#authors) downloads atm the Author File.

Thanks for noticing, I will have a solution soon.

> Can't clone the repo normally, thanks to diaspora*.yml

Which OS? Is that on Windows? As a workaround, yout could try to change the value of your [`core.protectNFTS` git setting](https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreprotectNTFS) to `false` (`git config --user core.protectNTFS false`). But I think the `*` characted should be trimmed during import in https://github.com/nodiscc/hecat/blob/master/hecat/utils.py#L15, for consistence - I will look into it. However I cannot guarantee that another contributor won't commit a filename that is incompatible with NTFS in the future.

**Comment by Rabenherz112** [comment]
> The non-free page will only be rendered in the markdown-based list (there is a link to it in the header of https://nodiscc.github.io/awesome-selfhosted-html-preview). 

Ah, I completely missed that.

> What about using a tooltip when hovering the mouse over the date? The tooltip would have the text 'Date of last update' or when rendered in range/red should have the tooltip 'Date of last update - this project has not received any update since {6,12} months!'

It seems like a promising suggestion, especially considering the existing system for Anti-features. Additionally, altering the text when it turns orange/red would help users comprehend the reason for the different colours. However, I'm slightly unsure about how many users will actually notice this change. Personally, while scrolling through the list, I wasn't aware that the anti-features tag had tooltips, as none of the other tags do. The idea of having tooltips for this specific tag didn't occur to me at the time.

> Is that on Windows? If so, you can try the workaround.

Indeed, it is Windows. I came across the workaround you mentioned online, but I must admit I'm not fond of disabling this protection.

> But I think the * character should be trimmed during import. However I cannot guarantee that another contributor won't commit a filename that is incompatible with NTFS in the future

That would be nice. I comprehend that it's something that might happen again in the future, and if it does, we'll likely need to deal with it once more. From what I can tell, occurrences like this are relatively rare.

**Comment by nodiscc** [comment]
> tooltip when hovering the mouse over the date

done https://github.com/nodiscc/hecat/commit/3c3d286f745cc161eee3354f8bb25d2ef930ee6d

> when rendered in orange/red should have the tooltip 'Date of last update - this project has not received any update since {6,12} months!'

Will be done later, the tooltip is sufficient for now.

> Can't clone the repo normally, thanks to diaspora*.yml, the * character should be trimmed during import

fixed https://github.com/nodiscc/hecat/commit/60760cd9053c0dc9d2e4aff13ca4b6072a8c7523

> The Author Button downloads atm the Author File

fixed https://github.com/nodiscc/hecat/commit/fefbd7c48da45259d17d8b05030a0e245664849a, https://github.com/awesome-selfhosted/awesome-selfhosted-data/commit/2322f5ce55150d5994a3dc1766e256b0f87fb2ff

-> preview updated https://nodiscc.github.io/awesome-selfhosted-html-preview/

**Comment by Rabenherz112** [comment]
Now this is annoying, I still have issuing when I am trying to clone the repo, now however because of `:`.
I will just use the fix, to continue testing,

**Comment by Rabenherz112** [comment]
Could you please take a moment to check whether PR #36 was created by me? I want to make sure I didn't misunderstand anything (and understand who that all should work). I got a little confused with the `depends_3rdparty` field, as it is set to true by default, but the description says it is false by default.

By the way, I must admit that creating a PR is much easier now. The provided template makes the process a lot smoother. 👍

**Comment by nodiscc** [comment]
>  I still have issues when I am trying to clone the repo, now however, because of `:`

Should be fixed in https://github.com/awesome-selfhosted/awesome-selfhosted-data/commit/59ed57f99328bbfe70bfc5ee660e448f2be612e8 (https://github.com/nodiscc/hecat/releases/tag/1.0.1) + https://github.com/awesome-selfhosted/awesome-selfhosted-data/commit/30fa8ee1d1eba57688833233afc880985bdcdcf6 (reimport data)

> Could you please take a moment to check whether PR https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/36 was created by me? 

Look good, all syntax checks passing as well (https://github.com/awesome-selfhosted/awesome-selfhosted-data/actions/runs/5648100392/job/15299667081?pr=36)

>  I got a little confused with the depends_3rdparty field, as it is set to true by default, but the description says it is false by default.

It is shown in the template with the value `true`, however if the software does not depend on 3rd party services, this field should be removed (as it defaults to `false`) - this is what you did and it is correct. But you are right that it can be confusing, maybe the template could make this more explicit.

Thanks

**Comment by nodiscc** [comment]
Merged, should be built in a few moments (https://github.com/awesome-selfhosted/awesome-selfhosted-data/actions/runs/5648280194/job/15300165745) and pushed to https://github.com/nodiscc/awesome-selfhosted and https://nodiscc.github.io/awesome-selfhosted-html-preview/

**Comment by Rabenherz112** [comment]
> Should be fixed in 59ed57f (https://github.com/nodiscc/hecat/releases/tag/1.0.1) + 30fa8ee (reimport data)

Works like a charm, thanks!

> Merged, should be built in a few moments 

Looks also good, also good to know that everything was good in the PR itself.

> actually I think there is a small bug/corner case, and we won't

😮

After implementing the last changes related to the Author File and other aspects, the website looks even better. I couldn't detect any other bugs. Again, you did an outstanding job! The only question I have is whether it was a deliberate decision to display `?` for Projects where the Metadata cannot be retrieved (Projects not from GitHub), instead of simply not showing the tag at all?

**Comment by nodiscc** [comment]
> whether it was a deliberate decision to display ? for Projects where the Metadata cannot be retrieved (Projects not from GitHub), instead of simply not showing the tag

Yes it was deliberate, it clearly shows that we are missing this information - it can still be added manually, or automation tools can be improved to retrieve it for non-Github projects.

**Comment by nodiscc** [comment]
Moved from https://github.com/awesome-selfhosted/awesome-selfhosted/pull/4027

> What are your plans for handling the migration itself?

- [x] setup CI in https://github.com/awesome-selfhosted/awesome-selfhosted-data to push the markdown version to a 
temporary/testing repo (https://github.com/awesome-selfhosted/awesome-selfhosted-testing?)
- [x] setup CI in https://github.com/awesome-selfhosted/awesome-selfhosted-data to push the HTML version to a temporary/testing repo (https://github.com/nodiscc/awesome-selfhosted-html-preview), use github pages to serve the preview website
- [x] **test** and simplify contribution process, test automatic generation/automation/CI/CD
- [x] get a cheap VPS (or possibly reuse my existing VPS), setup a static site virtualhost, setup cron to pull the new version of the HTML repo every hour or so
- [x] get a domain name, point it at the VPS (or alternatively use github pages for hosting [[1]](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site))
- [ ] transfer https://github.com/nodiscc/awesome-selfhosted-html-preview to https://github.com/awesome-selfhosted/awesome-selfhosted-html, update VPS cron job accordingly
- [ ] switch https://github.com/awesome-selfhosted/awesome-selfhosted-testing for https://github.com/awesome-selfhosted/awesome-selfhosted as target for the markdown page deployment
- [x] update https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/markdown/header.md with links to both markdown and HTML versions of the list https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/54
- [ ] update the link in the https://github.com/awesome-selfhosted/awesome-selfhosted project description/sidebar to point to the HTML site
- [ ] update https://github.com/awesome-selfhosted/awesome-selfhosted/blob/master/.github/CONTRIBUTING.md and https://github.com/awesome-selfhosted/awesome-selfhosted/blob/master/.github/PULL_REQUEST_TEMPLATE.md to point to the new contribution process on https://github.com/awesome-selfhosted/awesome-selfhosted-data
- [ ] migrate open issues and PRs from https://github.com/awesome-selfhosted/awesome-selfhosted/issues to awesome-selfhosted-data manually (or ask the original poster to resubmit)
- [ ] start maintaining/curating https://github.com/awesome-selfhosted/awesome-selfhosted-data

**Comment by nodiscc** [comment]
**https://awesome-selfhosted.net** is now live.

Please check the [release announcement](https://github.com/awesome-selfhosted/awesome-selfhosted/releases/tag/1.0.0) for more details.

**Comment by Rabenherz112** [comment]
@nodiscc Congration on the release of the Website and migration to the new repo. 🎉🎉

**Comment by Rabenherz112** [comment]
Also, if I spot it correctly, aren't we at the moment still missing the `needs to mature`, `feedback needed` and `reviewers wanted` labels?

**Comment by nodiscc** [comment]
Correct, I just added these labels, thank you. https://github.com/awesome-selfhosted/awesome-selfhosted-data/labels

**Comment by kana800** [comment]
Hey there,

This might be totally unwanted but I would like the link to open in a new tab instead of opening in the same tab; I know I can just configure it in the my browser itself

If you guys are fine with it I wouldnt mind just do a simple PR.  

Cheers

**Comment by nodiscc** [comment]
> I would like the link to open in a new tab instead of opening in the same tab

Most users will not want this (including me, I hate when websites do this - let me manage my tabs). You can open links in new tabs quite simply, using ctrl+click or middle-click, or two-finger-tap on touchpads, or right-click -> open in new tab, or configure your browser to always open links in new tabs...

**Comment by kana800** [comment]
> Most users will not want this (including me, I hate when websites do this - let me manage my tabs). You can open links in new tabs quite simply, using ctrl+click or middle-click, or two-finger-tap on touchpads, or right-click -> open in new tab, or configure your browser to always open links in new tabs...

Understandable : ) , I will configure it on my browser

**Comment by rauenzi** [comment]
Just wanted to drop this in here... I personally much prefer the markdown version of the list over the html version because (to me) the html version wastes a lot of space and makes it harder to see more than just a couple projects a once. So I have a bit of a design proposal to change that but I didn't want to just try to go implement it without gathering thoughts from you guys here.

Currently, there are 3 lines of "tags" for each project. The first line is really moreso buttons, the second line is tags, the third is a category tag. My proposal would be to move the button-tags to be a part of the heading itself, and collapse the rest of the tags into a single line. I think having the tags on a single line is okay and does not hurt visibility or the ability to distinguish them because they are already color-coded and have their own icons. I would also personally adjust the font-size and padding a bit. Here's how it might look:

![firefox_IwOMh3AtCQ](https://github.com/awesome-selfhosted/awesome-selfhosted-data/assets/6865942/30ec2cb4-ca79-4c0e-8129-d11096be92f3)

In the screenshot, `Adguard Home` and `admidio` are untouched. `Aimeos` is just making the layout adjustments I described with no changes to the styling.  `Adminer` changes both the layout and applies the font-size and padding changes I described. To me, that makes better use of the space and it's still easy to see and distinguish everything. The tags can even wrap to a new line if many are added and it will still look fine.

What are your thoughts on this?

The only other ideas I had to improve the readability would be to do a media query for larger screens/resolutions and increase the width of `.content`, it's quite small on my screen/resolution. Or, to reduce the saturation of the tag colors because currently they pop a lot and draw the eye away quite heavily from the actual title/description of each project.

**Comment by Rabenherz112** [comment]
I personally agree. When examining your example, it becomes apparent that it optimizes space utilization quite effectively. Additionally, it has the nice side effect of providing clearer delineation for the buttons, used for accessing the project's additional information (Website, Source Code, Demo, and Client), separating them from the rest of the informational labels

So overall I would also like to see such a design change (like done in your example with `Aimeos`)

**Comment by nodiscc** [comment]
> - move the button-tags to be a part of the heading itself
> - collapse the rest of the tags into a single line
> - adjust the font-size and padding a bit

I agree with the suggested changes.

They should be implemented in https://github.com/nodiscc/hecat/blob/master/hecat/exporters/markdown_multipage.py. Keep in mind that the website should remain usable on small displays (phones). I am unlikely to find the time to work on this, but would accept a pull request that implements these changes (ideally, one change by PR). Website generation can be tested by cloning https://github.com/nodiscc/hecat, and running `make clone_awesome_selfhosted test_export_awesome_selfhosted_html` inside the repository.

**Comment by rauenzi** [comment]
Awesome! I should hopefully have some time this week to work on this, but I figured I would check-in before opening any PR(s). I'll definitely make sure that any changes I make will remain accessible to all resolutions.

**Comment by kokomo123** [comment]
It seems that some github action runs need to be rerunned due to them being cancelled by the latest action being pushed through due to some hierarchy that's implied. 
![image](https://github.com/awesome-selfhosted/awesome-selfhosted-data/assets/70863536/fa0982de-9740-4308-a6f1-2ad10dd38f28)
does the update-metadata job solve these kinds of issues? just confirming

**Comment by nodiscc** [comment]
> some github action runs need to be rerunned

Nope, the `cancelled` workflows are caused by a more recent commit being pushed to `master` before the build for the previous commit was completed. See [`concurrency:`](https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/.github/workflows/build.yml#L9) [[1]](https://docs.github.com/en/actions/using-jobs/using-concurrency).

We don't want concurrent build jobs on the same branch, only the latest revision should be built - if a workflow is already running when a new commit is pushed to `master`, it is cancelled, and a new workflow run starts for the latest commit instead.

**Comment by kokomo123** [comment]
Thanks for the explanation then, so it'll still all be merged to the website. 💯

**Comment by Rabenherz112** [comment]
I noticed a recurring pattern in the project descriptions during my recent search on the website for a specific piece of software. Many include terms like 'open-source', 'free', 'self-hosted' and so forth, which feels redundant. This is an open-source list; of course, projects listed here will be open-source and, therefore, also free to use. Additionally, some projects reiterate their name in the description, for instance, 'XYZ is ...'

Here's an example for clarity:
```
+ Good Example
Feedpushr
An RSS aggregator with the capability to transform and forward articles to various outputs. It's a single binary, extendable through plugins.

- Bad Example
Metabase
An easy, open-source method for everyone in your company to ask questions and learn from data.
```
We have previously re-written similar descriptions (refer to #419, #423, etc.), but we haven't formally documented this practice or discussed it (at least, I couldn't find any issue regarding this; feel free to point me to the discussion if he had one before).

I'm interested in knowing your perspective on this matter. In my opinion, software descriptions should focus more on accurately defining the project rather than promotional language. This would include avoiding terms like "awesome", "free", "open-source", "easy", etc., unless they add substantial value to the description. What are your thoughts?

**Comment by nodiscc** [comment]
> Additionally, some projects reiterate their name in the description, for instance, 'XYZ is ...'

I agree with this, and it has been mentioned in https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/408#issuecomment-1837151273:

>> There could be some sort of guideline about how to phrase descriptions too (for example Minimalist text adventure game would be preferred to A minimalist text adventure game or $PROJECT is a minimalist text adventure game)

> Many include terms like 'open-source', 'free', 'self-hosted' and so forth, which feels redundant

Also agree.

What we need is an additional section `Other guidelines` or similar in https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/CONTRIBUTING.md, which lists lists these "soft" guidelines - I'm not sure we should add these directly to https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/.github/PULL_REQUEST_TEMPLATE.md, as the list of requirements could become overwhelming for potential contributors, and make the PR checklist less effective.

**Comment by Rabenherz112** [comment]
> Should we add this to the Pull request template

I don't think this would be useful. These guidelines are only relevant for a small fraction of additions but would significantly increase the number of requirements, resulting in overly complex checks for people who want to add new software.

I would align more with your suggested approach of introducing an extra section in the Contributing, or we could consider a separate 'review' document. This could detail these specific requirements, leaving it to the reviewer's discretion to identify and address any additional requirements. If created in a tutorial-like format, it could also assist new users who want to review PRs, by having one central place for all that information without the need to search through numerous unrelated issues in both the current and older repositories.

**Comment by nodiscc** [comment]
> I would align more with your suggested approach of introducing an extra section in the Contributing

Proposed changes in https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/500

> or we could consider a separate 'review' document.

I think we should add another section `Reviewing additions` to CONTRIBUTING.md and mention that it is an effective way to contribute (I am extremely grateful to people who help with this task). This boils down to:
- ensuring that the syntax is conform (which is indicated by the results of automated checks)
- ensuring that the project is not already listed on  awesome-sysadmin, staticgen.com, staticsitegenerators.net, dbdb.io
- ensuring that the software actually fits in the listed categories/`tags` (refer to the tags descriptions for the actual purpose of each tag)
- Ensuring that the list of languages/platforms actually matches the provided installation instructions
- Ensuring that the license actually matches the project's license, and that it does not have hidden dependencies on non-Free/closed-source components
- Ensuring that the software actually provides _[network services](https://en.wikipedia.org/wiki/Network_service) and [web applications](https://en.wikipedia.org/wiki/Web_application) which can be hosted on your own server(s)_ (e.g. purely local command-line/desktop/mobile applications are out of scope)
- Ensuring that the first release is actually 4 months old
- I probably forgot some things

**Comment by nodiscc** [comment]
For the first time, automated checks for unmaintained projects are passing ![](https://github.com/awesome-selfhosted/awesome-selfhosted-data/actions/workflows/check-unmaintained-projects.yml/badge.svg).

Thanks to everyone involved in cleaning up the list.

**Comment by nodiscc** [comment]
- [ ] Another item that would speed up reviews, is having prepared reply templates for common problems. A frequent cause for additions being rejected, is the project not having any releases yet.

Here is a draft for replying to such PRs. We should probably refine it a bit based on previous hand-crafted replies.

> Hi, thanks for your contribution.

> However, there are no tagged releases for this project. Our guidelines require that _ Any software project you are adding was first released more than 4 months ago._ I encourage you to create a release now and/or a simple [changelog](https://keepachangelog.com/en/1.1.0/) that will help users keep track of changes in the software (especially breaking changes or changes requiring configuration tweaks), and will allow administrators to install a known working, unchanging version (as opposed to always installing the latest development version).

> Once this is done, the project may be resubmitted to awesome-selfhosted when the first release reaches the age of 4 months.

> Thanks for understanding, and good luck with this project.

**Comment by Rabenherz112** [comment]
I agree with having reply templates for common problems. This wouldn't only speed up reviews but also ensure consistency across reviews from different people. But where should such templates be stored? 

I like the idea of encouraging best practices, like having a changelog, even though I think we should make it clearer that this is just advice, not a must-do.

Here's my slightly changed suggestion:
```
Hi, thanks for your contribution.

Currently, this project doesn't have any tagged releases. Our guidelines require that _Any software project you are adding was first released more than 4 months ago._ This count initiates only after a release has been created to ensure users need not rely on the latest development version to use the project.

We also recommend maintaining a simple [changelog](https://keepachangelog.com) and/or using [semantic commit](https://www.conventionalcommits.org) messages that will help users in tracking software changes, particularly those that are breaking changes or require configuration modifications.

Once a release has been created and the project has matured for at least 4 months, the project may be resubmitted to awesome-selfhosted when the first release reaches the age of 4 months.

Thanks for understanding, and good luck with this project.
```

Other Ideas for template messages:

- Project has releases but the first release is not 4 months old (this could be modified from the template for missing releases)
- Project isn't an end-user application with a web interface (SDK, command-line tool, and so on)
- Project is already featured on awesome-sysadmin or one of the other lists

**Comment by Ki-er** [comment]
Been a while since I've done prs for this list due to other commitments but Im back now!

I agree with the consistent messages, the only thing is that we can make sure is that they don't come across as to robotic and unapproachable.

**Comment by nodiscc** [comment]
It is very kind of some people to monitor the availability of https://awesome-selfhosted.net/ for me, however please refrain to issue full `GET` requests on the main page every few minutes because that will amount to a considerable amount of traffic, and will get flagged as abuse and result in a ban. `HEAD` requests should be fine, but rest assured that monitoring and alerting are already in place on my side. Thanks

![image](https://github.com/awesome-selfhosted/awesome-selfhosted-data/assets/3950390/4300a3f8-d684-488a-b1dd-5cae9bfd494f)

```
awesome-selfhosted.net:443 89.xxx.xxx.xxx - - [21/Jan/2024:16:54:32 +0100] "GET / HTTP/1.1" 200 6368484 "-" "Uptime-Kuma/1.23.11"
awesome-selfhosted.net:443 89.xxx.xxx.xxx - - [21/Jan/2024:16:59:32 +0100] "GET / HTTP/1.1" 200 6368484 "-" "Uptime-Kuma/1.23.11"
awesome-selfhosted.net:443 89.xxx.xxx.xxx - - [21/Jan/2024:17:04:33 +0100] "GET / HTTP/1.1" 200 6368484 "-" "Uptime-Kuma/1.23.11"
awesome-selfhosted.net:443 89.xxx.xxx.xxx - - [21/Jan/2024:17:09:33 +0100] "GET / HTTP/1.1" 200 6368484 "-" "Uptime-Kuma/1.23.11"
awesome-selfhosted.net:443 89.xxx.xxx.xxx - - [21/Jan/2024:17:14:33 +0100] "GET / HTTP/1.1" 200 6368484 "-" "Uptime-Kuma/1.23.11"
awesome-selfhosted.net:443 89.xxx.xxx.xxx - - [21/Jan/2024:17:19:34 +0100] "GET / HTTP/1.1" 200 6368484 "-" "Uptime-Kuma/1.23.11"
awesome-selfhosted.net:443 89.xxx.xxx.xxx - - [21/Jan/2024:17:24:34 +0100] "GET / HTTP/1.1" 200 6368484 "-" "Uptime-Kuma/1.23.11"
```

**Comment by nodiscc** [comment]
The VPS hosting https://awesome-selfhosted.net/ will undergo a maintenance/migration operation by the hosting provider (Ionos) somewhere between 15/04/2024 and 10/05/2024. The website will be unavailable during the maintenance. Further details/ETA will be provided when the maintenance is about to start.

**Comment by kokomo123** [comment]
https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/837#issuecomment-2334652188
I think we should document all the precedents we have set just for reference, as I forgot about this one, otherwise eh, at least we have you nodiscc to remember precedents l.

**Comment by Rabenherz112** [comment]
I found that Github has finally added a saved replies feature that can be used functionally via the slash command. This might be quite useful for a few of us (we no longer need to search the templates and copy and paste them into the review).

When you go into your settings there is a new section named 'Saved replies' where you can add templates.
![https://cdn.ravshort.com/rEGU2/XiKuXUzI30.png](https://github.com/user-attachments/assets/cc1c5135-db52-4b42-8c7b-3e681576b70a)

**Comment by kokomo123** [comment]
I will definitely set this up, thank you raben!

**Comment by Rabenherz112** [comment]
How do we handle software contributions that merely port an existing application to another system (e.g., Dockerization), especially when the repository is neither owned nor controlled by the original creator?

I am asking because I just saw #1027 which kinda goes in that direction

**Comment by nodiscc** [comment]
> How do we handle software contributions that merely port an existing application to another system (e.g., Dockerization)

I don't think we should approve these, it will open the gates to a flood of custom builds, third-party docker images... a quick search returns at least 2 other docker image projects for Anki [[1]](https://github.com/ankitects/anki/pull/2798) [[2]](https://github.com/ankicommunity/ankicommunity-sync-server), so I'd prefer if these solutions could be considered first:
- contributing the OCI image build procedure to the upstream project (Dockerfile, documentation)... Anki devs were apparently open to contributions in the issue linked above
- linking to the third-party docker image build from the upstream's project documentation

I will have a look at https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/1027

**Comment by armchairancap** [comment]
> I would align more with your suggested approach of introducing an extra section in the Contributing, or we could consider a separate 'review' document.

About that some of (completed) PRs seem to be waiting for more reviewers for over 30 days. (Disclosure: my PR happens to be one such PR (hence this comment), but I imagine other contributors in similar situation must be wondering as well.)

Rather than ask in the PR, I tried to figure out what `reviewers wanted` means and how a reviewer decides to go to PRs and take a look. The how-to-contribute [doco](https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/CONTRIBUTING.md#no-tagged-releases) doesn't explain that, though. 

I don't mind waiting, but I do wonder how does a PR author "motivate" reviews of PRs that are "ready to go"?

**Comment by nodiscc** [comment]
> what reviewers wanted means

It means that anyone (and I mean _anyone_ with a Github account) interested in seeing the PR move forward can use the review tab, check if everything is in order, and click `Approve` (or request changes).

There is no definition of an "official" reviewer, anyone can contribute to the review process.

> how a reviewer decides to go to PRs

Based on:
* free time
* interest in the submission, relative to other PRs (if I don't have much personal interest in a submission and there are more pressing things to do, I will focus on these other things)
* PRs with several approvals are also likely to attract my attention.

> how does a PR author "motivate" reviews of PRs that are "ready to go"

You can ping/send a reminder in the PR comments if you think it's been inactive for too long (30 days is too long).

**Comment by Ki-er** [comment]
To add to the above, we get lots of PRs, if a project is well documented so that we can check it easily and accurately, this is also likely to get attention by reviewers, hence the anybody can review. If a project is convoluted and complicated to where someone with basic self hosting knowledge (i.e me) cant review it easily, then it is likely to be missed until someone with more knowledge, (i.e nodiscc) can review it with their expertise. 

This is why we it is important to make sure projects you add have documentation, new releases and are maintained so that we can review quickly and accurately, it is also important that when we make suggested changes, they are acted upon / or disputed against so that the PR doesn't sit dormant.

**Comment by armchairancap** [comment]
Thank you both.

> It means that anyone (and I mean anyone with a Github account) interested in seeing the PR move forward can use the review tab,

Yeah, that was another question I had. I obviously didn't want to attempt to review my own PR, but I wondered who may choose to do it. 
Maybe add a note to the review process with an invitation to community members to help out with reviews as well (if qualified)? I may review 2-3 PRs (other than mine) to help out.

**Comment by Ki-er** [comment]
We welcome new reviewers, even if you don't feel qualified, there are simple things that you can check for that are often missed for example:
 - depends 3rd party: false - this should be removed since it's optional and defaulted as false
- Check for releases, there should be atleast one and it should be atleast 4 months old
- check for the mention of Docker, this is sometimes missed in the platform and can be a simple addition suggestion.

Any questions you can always ask! The awesome selfhosted is a community list made for the community and maintained by the community!

**Comment by khansa01p** [comment]
#isntall #fix(Optional)
#download”package;

**Comment by Rabenherz112** [comment]
@nodiscc do you plan to add additional maintainers (members) in the future, or do you prefer to keep the project more controlled?
If you are open to additional maintainers, what requirements would there be?

**Comment by nodiscc** [comment]
@Rabenherz112 added you as maintainer. Thank you for your continued help with the project. (with great power comes great responsibility, blah, blah, if in doubt about something seek more input from me or other contributors, keep the true FOSS spirit **strong**)

🚀

**Comment by kokomo123** [comment]
Congratulations to kier on making maintainer as well 🎉

**Comment by Rabenherz112** [comment]
@nodiscc Congratulation on the 10 Year Anniversary of Awesome Selfhosted.

```
commit beb45bce4c03971155074f0289ba425f5990e11d
Author: Kickball <ed.kickball@hotmail.com>
Date:   Mon Jun 1 03:33:17 2015 +0100

    Initial commit
```

**Comment by kokomo123** [comment]
<img width="331" height="188" alt="Image" src="https://github.com/user-attachments/assets/c98c6afc-91ad-46f4-82f1-7f9c0088c7c2" />
Interesting how github silently rolled this out, letting read only people be added under +1 or multiple category if they dont have read / write perms, but eh, its fine nonetheless, at least im still being counted :P 

EDIT: https://github.blog/changelog/2025-08-14-clearer-pull-request-reviewer-status-and-enhanced-email-filtering/ found the changelog, well it seems fine to keep it organized

**Comment by clusterzx** [comment]
https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/1407#issuecomment-3298746585

@kokomo123 Ok, lets continue to talk here 😄 
I mean yeah, you could call it an addon to paperless-ngx. But I mean its now so big in functionality with RAG Server, automations and many other features to only call it an "addon" now. 

The solution has now over 4.2k Stars and 1.5million downloads. I dont get it why it should not be listed in at least the "Document Management" category.

**Comment by kokomo123** [comment]
@nodiscc @Rabenherz112 Just mentioning you here for this convo if you wish to chime in. We can certainly reconsider adding this back to awesome-selfhosted, I am not opposed to that, I just don't know now in September what the consensus is.

**Comment by nodiscc** [comment]
> Ok, lets continue to talk here [...] paperless-ngx

Please don't, open a specific issue instead

**Comment by clusterzx** [comment]
> > Ok, lets continue to talk here [...] paperless-ngx
> 
> Please don't, open a specific issue instead

I was told to come over here... nevermind. _*Sigh*_  I will open up an issue.

**Comment by kokomo123** [comment]
> > Ok, lets continue to talk here [...] paperless-ngx
> 
> Please don't, open a specific issue instead

I apologize for not leading to making a new issue.

**Comment by clusterzx** [comment]
> > > Ok, lets continue to talk here [...] paperless-ngx
> > 
> > 
> > Please don't, open a specific issue instead
> 
> I apologize for not leading to making a new issue.

No problem at all 😆 I think managing such a big repo is not that easy

**Comment by nodiscc** [comment]
Sorry for the confusion. This thread is for general discussion around the project, it can be used to determine possible areas of interest/investigation, but once a specific issue has been identified... well a separate issue is preferred.

**Comment by azc5OQ** [comment]
Hey, just had a question

Would you consider this project to be worthy of adding to awesome selfhosted?
https://github.com/azc5OQ/lemon-chat

I already tried to do that here
https://github.com/awesome-selfhosted/awesome-selfhosted/pull/3419

but it was never added for some reason

**Comment by kokomo123** [comment]
> Hey, just had a question
> 
> Would you consider this project to be worthy of adding to awesome selfhosted? https://github.com/azc5OQ/lemon-chat
> 
> I already tried to do that here [awesome-selfhosted/awesome-selfhosted#3419](https://github.com/awesome-selfhosted/awesome-selfhosted/pull/3419)
> 
> but it was never added for some reason

the right repository is here, feel free to readd a PR here! The awesome-selfhosted repo is only used for markdown purposes. 

EDIT: also seems like you forgot to correct a suggestion nodiscc was suggesting, thus we closed the PR.

**Comment by kokomo123** [comment]
I am currently trying to work through some backlog items in issues, so apologies for the PR spam incoming. 

EDIT: We should probably also look at some old PRs as well if anyone is willing to assist in that regard.

**Comment by nodiscc** [comment]
Using LLMs to submit PRs **and** grossly disregarding/not reading contribution guidelines and wasting maintainers time is now grounds for an immediate ban without further explanations.

https://github.com/awesome-selfhosted/awesome-selfhosted-data/pull/2033 is the last to date, but there has been an influx of such PRs recently (common pattern: vibe-coded projects with 20 commits and 1 week of existence by the author of the PR themselves). At least instruct your coding assistants to keep the PR template.

**Comment by DatAres37** [comment]
There seems to be a relatively new license, that is used by some projects in the list, for example [Hemmelig](https://github.com/HemmeligOrg/Hemmelig.app?tab=License-1-ov-file): [O'Saasy License Agreement](https://osaasy.dev/).
Are softwares with this license in the scope of Awesome-Selfhosted? It allows selfhosting, but basically only in private.

**Comment by Rabenherz112** [comment]
> Are softwares with O'Saasy License Agreement in the scope of Awesome-Selfhosted?

Taking a quick look the the O'Saasy License it seems to be not libre-free (it restricts use by prohibiting licensees from competing with the licensor via SaaS). Projects using this license are **therefore not in scope for the main list** but can be listed in the [non-free section](https://github.com/awesome-selfhosted/awesome-selfhosted/blob/master/non-free.md) (which is only accessible as markdown version).

( I will push a PR later to update Hemmelig, which is still listed here as an MIT license)

**Comment by kokomo123** [comment]
tfw these bots are outta control. The difference is quite hard, some are actual humans who then use LLMs to make the PR template, quite possibly still a bot? Some ask, "why did you close", others are just bots.. hrm 💀 Thank you AI for making my life a bit harder to maintain a list 👎🏻

**Comment by Rabenherz112** [comment]
> some are actual humans who then use LLMs to make the PR template

Tbh, whether they are human or not is irrelevant at that point. If someone can't spare 2 minutes to actually read and fill out the PR template and instead just lets their LLM hallucinate through it, I am not going to spend *my* time reviewing that PR. Reviewing takes a lot longer than the 2 minutes it would take to make yourself familiar with our contribution workflow (which in my mind is well enough documented).

It is a basic matter of respecting maintainers time. If a contributor can't be bothered to put in the bare minimum effort and follow basic guidelines, why should we invest our time making sure it's correct and valid?

**Comment by nodiscc** [comment]
> If someone can't spare 2 minutes to actually read and fill out the PR template and instead just lets their LLM hallucinate through it, I am not going to spend my time reviewing that PR

Exactly my point.

Aside from the PR template not being respected, most if not all of these submissions are weekend projects written by LLMs too (everything, the code, the docs...). Don't get me wrong, I also use LLMs to write non-critical software for personal use, but this list is not the right place to share such projects.

Another potential useful metric (in addition to obvious LLM tells everywhere in these repos) is the number of SLOC changed in the last few days [[1]](https://gist.github.com/nodiscc/febd128c1a91f2d588961cf91287abfd). How can any maintainer or team commit 15k SLOC changes in 3 days while keeping good review/test discipline? It is a serious clue that these projects are entirely vibe-coded with little oversight from their "maintainers".

The next question is, do we want to allow such projects in the list? Even if the submission itself is "correct".

The 4-months-since-first-release rule has kept us relatively safe from these submissions until now. Adding "quality" guidelines was discussed in the past and is a complex problem that was never fully solved - but the current situation makes the issue both more complex, and more pressing.
 - should we add an "AI-assisted development" anti-feature flag? I expect the majority of software in the next few years to be developed with LLM assistance at some degree and this is not necessarily bad, LLMs _do_ have good use cases even in properly maintained projects (bug discovery, security analysis, speeding up prototyping/initial implementation, the script I shared above was written by qwen3.5 on llama.cpp...). So not a very useful tag.
- should web add a "full AI-developed" anti-feature or outright refuse these projects? I like the idea, but the type of people who submit such software are not going to voluntarily disclose this, so we're back to square one.
- Increase the delay from 4 months to 6 months? This will give us a bit more time until this current trend subsides... but I don't know if it ever will.

**Comment by Rabenherz112** [comment]
Moving this to a seperate Issue, since I can see this already getting big

**Comment by kokomo123** [comment]
Moving my comment to over there, deleting original here.

**Comment by kokomo123** [comment]
Trying to clean out some PRs from our queue rn, if ya'll could look at the one month old ones once more, im pinging some ones that should be fine to nodiscc and you raben.

**Comment by kokomo123** [comment]
I just wanted to say, thank you raben, nodiscc, kier, kickball for allowing me to play a bigger role in awesome-selfhosted. I won't let you down 💯 🙏🏻 ❤️ (sorry for pings but just wanted to say it).

**Comment by nodiscc** [comment]
Granted write access to @kokomo123
Thanks for your continued help. Use responsibly, if in doubt seek feedback from other contributors, more importantly: have fun, it's not a job, there are no obligations whatsoever, letting things fly for a bit is fine, not replying is fine, only approving projects you find interesting and high quality is fine. In fact the process is **slow** by design, the list should not contain every existing project, don't let anyone pressure you into doing more or doing faster, etc.

```bash
# list is already chonky
$ awesome-selfhosted-data (master =)$ ls software/|wc -l
1314
# list is old
$ awesome-selfhosted-data (master =)$ git log --oneline |wc -l
9184
awesome-selfhosted-data (master =)$ git show beb45bce|grep Date
Date:   Mon Jun 1 03:33:17 2015 +0100
# and doing fine
$ awesome-selfhosted-data (master =)$ cat AUTHORS | wc -l
1706
```

Thanks to everyone involved

**Comment by kokomo123** [comment]
Well, another update in the mix. Thank you for the membership to awesome-selfhosted @nodiscc 💀 Didn't expect this to happen today but hey im here

**Comment by nodiscc** [comment]
@kokomo123 you already were a member at the repository level, I added you at the organization level so you can block misbehaving users, see https://github.com/awesome-foss/awesome-sysadmin-data/pull/70

**Comment by kokomo123** [comment]
Ahhh, noted. well that makes more sense then 😆

**Comment by Ki-er** [comment]
Hi all, My PhD is about to be submitted so I plan on being more active here! Apologies for the ghostiness; always happy to drop in and provide my well-informed opinion!

**Comment by kokomo123** [comment]
<img width="141" height="30" alt="Image" src="https://github.com/user-attachments/assets/b9765290-bfc1-43c3-8b76-166e4bbf00be" /> New github observation 👀 it's now easier to see a "green checkmark" for approving stuff and how long since it's been for the PR to be updated like new activity

**Comment by Rabenherz112** [comment]
Back under 100 open issues... Yaaaay....
While speed reviewing them just now, I actually saw some good looking projects..
Probably need to reserve some time to actually go through them and create PRs for some them..
